docs/operate/get-started/setup-micro/_index.md

```diff
@@ -0,0 +1,404 @@
+
+1. Generate a new project skeleton from [this template](https://github.com/viamrobotics/micro-rdk/tree/main/templates/project):
+
+   ```sh { class="command-line" data-prompt="$"}
+   cargo generate --git https://github.com/viamrobotics/micro-rdk.git
+   ```
+
+   Follow the prompts:
+
+   - Which sub-template should be expanded?: `templates/project`.
+   - Project name: Your choice.
+   - MCU: `esp32`.
+   - Include camera module?: If you will use an `esp32-camera` or a `fake` camera as a component of your machine, select `true`.
+   - Machine cloud credentials: Paste in the machine cloud credentials you obtained in step 1
+   - Enter WiFi name: The name of the WiFi network for the ESP32 to connect to on boot. Note that you must provide a 2.4 GHz network.
+   - Enter the WiFi password: The password for the 2.GHz WiFi network.
+
+   Hit **Enter** to generate the project.
+   The CLI automatically initializes a git repository in the generated directory for version control and in case you want to use cloud build later.
+
+1. Navigate into the generated project:
+
+   ```sh { class="command-line" data-prompt="$"}
+   cd <path-to/your-project-directory>
+   ```
+
+1. Add any desired modules to the project by including them in the `dependencies` section of the `Cargo.toml` for the generated project:
+
+   {{< tabs >}}
+   {{% tab name="Local path dependency" %}}
+
+   While developing a module, you'll typically use a local path dependency to reference your module directory, for example:
+
+   ```toml {class="line-numbers linkable-line-numbers" data-line="3"}
+   [dependencies]
+   ...
+   my-module = { path = "../my-module" }
+   ```
+
+   {{% /tab %}}
+   {{% tab name="Git repository dependency" %}}
+
+   For modules hosted in a Git repository, specify the repository URL and optionally a specific commit (`rev`), branch (`branch`), or tag (`tag`), for example:
+
+   ```toml {class="line-numbers linkable-line-numbers" data-line="3"}
+   [dependencies]
+   ...
+   my-module = { git = "https://github.com/username/my-module.git", tag = "v1.0.0" }
+   ```
+
+   To use any of the [example modules](https://github.com/viamrobotics/micro-rdk/blob/main/examples/modular-drivers/README.md#example-modules) provided by Viam, use this line, specifying the correct Micro-RDK version that matches your development environment:
+
+   ```toml {class="line-numbers linkable-line-numbers" data-line="3"}
+   [dependencies]
+   ...
+   micro-rdk-modular-driver-example = { git = "https://github.com/viamrobotics/micro-rdk.git", rev = "v0.5.0", package = "micro-rdk-modular-driver-example", features = ["esp32"] }
+   ```
+
+   {{% /tab %}}
+   {{< /tabs >}}
+
+1. Compile the project:
+
+   ```sh { class="command-line" data-prompt="$"}
+   make build-esp32-bin
+   ```
+
+   The first build may be fairly time consuming, as ESP-IDF must be cloned and built, and all dependent Rust crates must be fetched and built.
+   Subsequent builds will be faster.
```

Comment:
```suggestion
   The first build can take a few minutes, as ESP-IDF must be cloned and built, and all dependent Rust crates must be fetched and built.
   Subsequent builds will be faster.
```
[s] provide a concrete timeframe, even if it's on the high side, since 'fairly time consuming' could take hours in the hardware world and I don't think this could take hours, even on dial-up!

Reply:
Subsequent builds also take several minutes :D I like the idea of being more concrete but not sure what the upper bound might look like-- @gvaradarajan ?